src/fontra/views/editor/panel-transformation.js

```diff
@@ -450,6 +579,452 @@ export default class TransformationPanel extends Panel {
+    const widthSelection = selectionBounds.xMax - selectionBounds.xMin;
+    const distributeSpacer = {
+      width: (widthSelection - effectiveDimentions.width) / (dictributeObjectCount - 1),
+      height:
+        (heightSelection - effectiveDimentions.height) / (dictributeObjectCount - 1),
+    };
+    return distributeSpacer;
+  }
+
+  _transformObject(
+    undoLabel,
+    layerGlyph,
+    objectBounds,
+    selectionBounds,
+    nextPosition,
+    distributeSpacer,
+    individualSelection,
+    editChanges,
+    changePath,
+    rollbackChanges
+  ) {
+    const { translateX, translateY } = this._getTranslationForObject(
+      undoLabel,
+      objectBounds,
+      selectionBounds,
+      nextPosition,
+      distributeSpacer
+    );
+
+    this._alignObjectEditBehaviour(
+      layerGlyph,
+      individualSelection,
+      translateX,
+      translateY,
+      editChanges,
+      changePath,
+      rollbackChanges
+    );
+  }
+
+  async _alignObjectsLayerGlyph(undoLabel) {
+    let { point: pointIndices, component: componentIndices } = parseSelection(
+      this.sceneController.selection
+    );
+
+    pointIndices = pointIndices || [];
+    componentIndices = componentIndices || [];
+    if (!pointIndices.length && !componentIndices.length) {
+      return;
+    }
+
+    const varGlyphController =
+      await this.sceneController.sceneModel.getSelectedVariableGlyphController();
+
+    const editingLayers = this.sceneController.getEditingLayerFromGlyphLayers(
+      varGlyphController.layers
+    );
+    const staticGlyphControllers = {};
+    for (const [i, source] of enumerate(varGlyphController.sources)) {
+      if (source.layerName in editingLayers) {
+        staticGlyphControllers[source.layerName] =
+          await this.fontController.getLayerGlyphController(
+            varGlyphController.name,
+            source.layerName,
+            i
+          );
+      }
+    }
+
+    await this.sceneController.editGlyph((sendIncrementalChange, glyph) => {
+      const editLayerGlyphs = this.sceneController.getEditingLayerFromGlyphLayers(
+        glyph.layers
+      );
+
+      const editChanges = [];
+      const rollbackChanges = [];
+      for (const [layerName, layerGlyph] of Object.entries(editLayerGlyphs)) {
+        const layerGlyphController = staticGlyphControllers[layerName];
+        const changePath = ["layers", layerName, "glyph"];
+
+        const { points, contours, components } = this._splitSelection(
+          layerGlyphController,
+          this.sceneController.selection
+        );
+
+        let selectionBounds = layerGlyphController.getSelectionBounds(
+          this.sceneController.selection
+        );
+
+        if (
+          (contours.length == 1 && !components.length && !points.length) ||
+          (components.length == 1 && !contours.length && !points.length) ||
+          (points.length < 1 && !contours.length && !components.length)
+        ) {
+          // if only one object is selected
```

Comment:
This sounds a little arbitrary and I'm not sure we should include this "single object" behavior. And if we do, I'm not sure it should behave like it does. It could also use the glyph bounding box?

Reply:
The 'single object' behaviour is similar to figma 'Select one object or layer':
https://help.figma.com/hc/en-us/articles/360039956914-Adjust-alignment-rotation-and-position#h_01HNBH55643QCPMYAQZHWSXPJY

Reply:
But then again, we don't really have the concept of "parent", and the box defined by lsb/rsb/descender/ascender is rather arbitrary. At some point we'll also have tsb/bst (yAdvance + verticalOrigin).

What makes sense for a layout app does not necessarily make sense in a font editor.

Reply:
I removed it. Done.